enhancements/network/infiniband.md

```diff
@@ -0,0 +1,123 @@
+---
+title: infiniband
+authors:
+  - "@zshi"
+reviewers:
+  - "@pliurh"
+  - "@dougbtv"
+  - "@s1061123"
+  - "@danwinship"
+  - "@squeed"
+  - "@dcbw"
+  - "@yrobla"
+approvers:
+  - "@knobunc"
+  - "@fepan"
+creation-date: 2020-05-06
+last-updated: 2020-05-06
+status: implementable
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Infiniband
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [X] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+RDMA (Remote Direct Memory Access) fabrics such as Infiniband and RDMA over
+Converged Ethernet (RoCE) are key to unlocking scalable performance for
+demanding Artificial Intelligence (AI) and Machine Learning (ML) applications.
+RoCE v2 is supported in SR-IOV Operator as Tech Preview since 4.3, This
+enhancement is to introduce native Infiniband support.
+
+## Motivation
+
+Enable high-performance message transmission via RDMA for distributed workloads
+such as AI/ML.
+
+### Goals
+
+- Provide native Infiniband support on pod additional network interfaces
+
+### Non-Goals
+
+- Support RoCE v2
+- Support GPUDirect RDMA
+- Support Ethernet over Infiniband such as IPoIB (IP over Infiniband)
+- Support Infiniband on pod primary network interface
+
+## Proposal
+
+Extend SR-IOV Operator to support both Ethernet and Infiniband devices.
+
+SR-IOV Hardware NIC is managed by SR-IOV Operator and will be provisioned
+with a new device type `infiniband` or `ib`.
+
+A new Container Network Interface (CNI) plugin `ib-sriov-cni` will be added.
+
+Infiniband support will only be available on bare metal deployment.
+
+#### Hardware
+
+The following Mellanox NICs will be supported:
+- Mellanox Technologies MT27700 Family [ConnectX-4]
+- Mellanox Technologies MT28908 Family [ConnectX-6]
+
+#### Infiniband SR-IOV CNI
+
+Infiniband SR-IOV CNI `ib-sriov-cni` is a 3rd party CNI plugin that moves
+SR-IOV Infiniband device to pod namespace and configures device attributes
+such as Infiniband GUID. It will be delivered in a separate image and copied
+to node CNI directory by a DaemonSet.
+
+#### SR-IOV Config Daemon
+
+SR-IOV Config Daemon, a sub component managed by SR-IOV Operator, will be
+enhanced to change device link type to Infiniband.
+
+#### SR-IOV Operator
```

Comment:
We shall also let the operator support generating the net-att-def for the ib-sriov-cni, right? And I think we shall also make a statement that the sriov-device-plugin has already support the sriov IB devices

Reply:
Good point, added .